WordPress/openverse
# Limit access for content moderators

## Fixes



Fixes #3640 by @sarayourfriend 

## Description



This PR
- changes all moderator access to just "view".

  This means they can only view the model instances from the admin view. Actions like edit, delete and make changes are not available to them. They can still file reports and take decisions on pending reports using the form within the single media item view in the admin as those are implemented through custom views.

- adds checks on the custom views for taking decision

  The decision models, and the corresponding through models, are created through the custom decision making view. This PR also adds a check to this view that only allows `POST` requests from people with `is_superuser` flag or "Content Moderators"-group membership.

## Testing Instructions


0. Check out the PR.
1. Re-init the API DB so that the old moderators group is replaced with new permissions.

   ```bash
   just down -v && just api/init
   ```

2. Login into the Django admin as the user "moderator" (password: "deploy"). You should not see any controls to edit or delete any models.
3. Go to the single media item view and try to file report and perform moderation. You should still be able to do both of these things.

## Checklist


      PRs) or the media properties generator (`./ov just catalog/generate-docs media-props`
      for the catalog or `./ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
